pkg/steps/pod.go

```diff
@@ -54,35 +58,12 @@ func (s *podStep) Run(ctx context.Context, dry bool) error {
 	}
 	image := fmt.Sprintf("%s:%s", s.config.From.Name, s.config.From.Tag)
 
-	pod := &coreapi.Pod{
-		ObjectMeta: meta.ObjectMeta{
-			Name: s.config.As,
-			Labels: trimLabels(map[string]string{
-				PersistsLabel:    "false",
-				JobLabel:         s.jobSpec.Job,
-				BuildIdLabel:     s.jobSpec.BuildId,
-				ProwJobIdLabel:   s.jobSpec.ProwJobID,
-				CreatedByCILabel: "true",
-			}),
-			Annotations: map[string]string{
-				JobSpecAnnotation:                     s.jobSpec.RawSpec(),
-				annotationContainersForSubTestResults: s.name,
-			},
-		},
-		Spec: coreapi.PodSpec{
-			ServiceAccountName: s.config.ServiceAccountName,
-			RestartPolicy:      coreapi.RestartPolicyNever,
-			Containers: []coreapi.Container{
-				{
-					Name:                     s.name,
-					Image:                    image,
-					Command:                  []string{"/bin/sh", "-c", "#!/bin/sh\nset -eu\n" + s.config.Commands},
-					Resources:                containerResources,
-					TerminationMessagePolicy: coreapi.TerminationMessageFallbackToLogsOnError,
-				},
-			},
-		},
+	pod, err := s.generatePodForStep()
+	if err != nil {
+		return err
 	}
+	pod.Spec.Containers[0].Image = image
```

Comment:
nit: why not passing the `image` and `containerResources` in `generatePodForStep()`